src/creature_states_pray.c

```diff
@@ -170,9 +168,15 @@ CrCheckRet process_temple_function(struct Thing *thing)
     return CrCkRet_Available;
 }
 
-short state_cleanup_in_temple(struct Thing *thing)
+short state_cleanup_in_temple(struct Thing *creatng)
 {
-  return _DK_state_cleanup_in_temple(thing);
+    struct Dungeon* dungeon = get_dungeon(creatng->owner);
+    if ( dungeon->creatures_praying[creatng->model] > 0 )
+       dungeon->creatures_praying[creatng->model]--;
+    else
+        ERRORLOG("No creature in temple to cleanup");
+    state_cleanup_in_room(creatng);
```

Comment:
this looks like a prime location for brackets to me. To make it absolutely clear we're not doing the cleanup on the 'else'